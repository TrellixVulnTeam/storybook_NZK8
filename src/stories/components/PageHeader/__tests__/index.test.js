import React from 'react';
import { shallow, mount } from 'enzyme';
import store from 'store';
import { PageHeader } from '../index';

describe('<PageHeader />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('snapshot', () => {
    const c = shallow(<PageHeader store={store} />);

    // eslint-disable-next-line jest/prefer-inline-snapshots
    expect(c).toMatchSnapshot();
  });

  it('should handle props.title', () => {
    const c = mount(<PageHeader store={store} title="Foobar" />);

    expect(c.find('._test-page-header-title').text()).toContain('Foobar');
  });

  it('should handle MenuIcon.onClick', () => {
    const c = mount(<PageHeader store={store} />);

    // Clicking on the menu icon should update the redux state `showMainMenu` property to true
    c.find('._test-menu-icon').simulate('click');

    expect(store.getState().showMainMenu).toStrictEqual(true);
  });
});
