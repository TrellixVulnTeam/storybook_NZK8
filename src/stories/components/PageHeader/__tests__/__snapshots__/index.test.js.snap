// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> snapshot 1`] = `
<styled__Wrapper>
  <test-file-stub
    className="_test-menu-icon menu-icon"
    onClick={[Function]}
  />
  <label
    className="_test-page-header-title"
  />
  <_default
    onClick={[Function]}
    size="xs"
    theme="cancel"
  >
    <test-file-stub />
  </_default>
  <span>
    <test-file-stub />
  </span>
  <span
    onClick={[Function]}
  >
    <DropdownMenu
      direction="bottom-left"
      display={false}
      header=""
      menuItems={
        Array [
          Object {
            "handleOnClick": [Function],
            "label": "View Profile",
          },
          Object {
            "handleOnClick": [Function],
            "label": "View Company Profile",
          },
          Object {
            "handleOnClick": [Function],
            "label": "Log Out",
          },
        ]
      }
      offsetY="30"
      width="160"
    >
      <test-file-stub />
    </DropdownMenu>
  </span>
</styled__Wrapper>
`;
